### PR TITLE
[OSD-18452] Adds metrics for lifecycle, errors, generic info

### PR DESCRIPTION
**Pull request by clcollins**
Adds metrics to track received webhooks, processing, errors, success and
uptime.

Fixes [OSD-18452](https://issues.redhat.com//browse/OSD-18452)

Signed-off-by: Chris Collins <collins.christopher@gmail.com>

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		MetricUptime,
+	}
+)
+
+func RecordUptime() {
+	for {
+		MetricUptime.Inc()
+		time.Sleep(time.Second)
+	}
+}
```
Comment (mjlshen) [inline pkg/metrics/metrics.go:184]: Something I've seen in other metrics is exposing a metric as the Linux epoch time it started. Then, uptime can always be calculated with the PromQL:

```
time() - compliance_audit_router_started
```

I think the main advantage would be putting less load on Prometheus as well as the node this pod ends up running on

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
+	// MetricHTTPSuccesses is the number of HTTP successes returned by the application
+	MetricHTTPSuccesses = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_http_successes",
+		Help:        "Number of HTTP successes returned by the application",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
```
Comment (mjlshen) [inline pkg/metrics/metrics.go:147]: What do you think about returning HTTP code as a label of this metric? Then you could slice and dice the data as you wanted with PromQL?

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+	// JIRA ISSUE CREATION FOR EVENTS
+
+	// MetricJiraClientCreateFailures is the number of failures to create a Jira client
+	MetricJiraClientCreateFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "compliance_audit_router_jira_client_create_failures",
+		Help: "Number of failures to create a Jira client",
+		ConstLabels: map[string]string{
+			"name": "compliance-audit-router",
+		}},
```
Comment (mjlshen) [inline pkg/metrics/metrics.go:71]:
```suggestion
		ConstLabels: CARPrometheusLabels},
```

**Review by mjlshen — COMMENTED** [review]
lgtm overall, just a couple questions

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (clcollins) [reply pkg/metrics/metrics.go]: Nice, that makes sense, and I discovered prometheus has a `process_start_time_seconds` already included, so no extra code is needed here for this.

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (clcollins) [reply pkg/metrics/metrics.go]: Great idea - replaced these two with `MetricHTTPResponses` in the code, and can use the `[]string{"code"...` bit as the label for the response code.

**Review by mjlshen — APPROVED** [review]
/lgtm

**Comment by clcollins** [comment]
/hold

Tests need to be fixed up

**Pull request merged**